Apply black

dalf opened the pull request:
## What does this PR do?

This PR is demo of [black](https://pypi.org/project/black/):
* first commit `black -l 120 -S searx searxng_extra` : reformat except the simple quote.
* second commit : `black -l 120 searx searxng_extra`: reformat including the simple quote.

## Why is this change important?

Normalize code format.

## How to test this PR locally?

## Author's checklist

* About PEP8 E203: see https://github.com/psf/black/issues/315
* If we use black, we can disable pycodestyle, and just reformat the code (and keep pylint). The tooling is not modified, so `make test` fails for now.
* there is an emacs package: https://github.com/wbolster/emacs-python-black


## Related issues

<!--
Closes #234
-->

dalf committed:
[mod] replace pycodestyle by black

"make test.black" checks for the code style
"make format.python" format the python code

return42 commented on searx/search/checker/impl.py:
```diff
@@ -25,13 +25,87 @@
+    'applet',
+    'frame',
+    'frameset',
 ]
```
Over all I think black formatting is good, but there are some exceptions like this one here.

In such cases we should use `# fmt: off` & `# fmt: on`.

return42 committed:
[mod] test.black: show issues (diff) reported by python-black

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 reviewed (approved):
LGTM / thanks!

@dalf See my comment and the patches I pushed on top of your branch.

return42 committed:
[format.python] disable py code formatting for some hunks of code

Disable the python code formatting from python-black, where the readability of
code suffers by formatting.

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[format.python] initial formatting of the python code

This patch was generated by black [1]::

    make format.python

[1] https://github.com/psf/black

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[fix] issues reported by pylint

Fix pylint issues from commit (3d96a983)

    [format.python] initial formatting of the python code

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>
